pkg/virt-config/virt-config.go

```diff
@@ -69,20 +72,41 @@ const (
 	DefaultVirtOperatorLogVerbosity                 = 2
 )
 
-// Set default machine type and supported emulated machines based on architecture
-func getDefaultMachinesForArch() (string, string) {
-	if runtime.GOARCH == "ppc64le" {
+type DefaultArch struct {
+	Architecture string
+}
+
+func NewDefaultArch(arch string) *DefaultArch {
+	return &DefaultArch{Architecture: arch}
+}
+
+// GetDefaultMachinesForArch set default machine type and supported emulated machines based on architecture
+func (d *DefaultArch) GetDefaultMachinesForArch() (string, string) {
+	if d.Architecture == "ppc64le" {
```

Comment:
Much better now without `runtime.GOARCH`. Maybe we can hide the strings inside a DefaultArch function to avoid copy-paste errors on the string values or define  constants?

Reply:
I am not sure if my understanding is correct. I write a simple code as following:
```
type defaultValue struct {
        machineType string
        machines    string
        ovmfPath    string
}
func (d *DefaultArch) GetDefaultValueForArch() (string, string, string){
        amd64Value := defaultValue {machineType: DefaultAMD64MachineType, machines: DefaultAMD64EmulatedMachines, ovmfPath: DefaultARCHOVMFPath}
        ppc64leValue := defaultValue {machineType: DefaultPPC64LEMachineType, machines: DefaultPPC64LEEmulatedMachines, ovmfPath: DefaultARCHOVMFPath}
        arm64Value := defaultValue {machineType: DefaultAARCH64MachineType, machines: DefaultAARCH64EmulatedMachines, ovmfPath: DefaultAARCH64OVMFPath}
        archDefault := map[string]retValue{"amd64": amd64Value, "ppc64le": ppc64leValue, "arm64": arm64Value}
        return archDefault[d.Architecture].machineType, archDefault[d.Architecture].machines, archDefault[d.Architecture].ovmfPath
}

var DefaultMachineType, DefaultEmulatedMachines, DefaultOVMFPath = NewDefaultArch(runtime.GOARCH).GetDefaultValueForArch()
```

Reply:
Something like this, but if you go into that direction, maybe more like this:

```
NewDefaultArch(runtime.GOARCH).IsPPC64()
NewDefaultArch(runtime.GOARCH).IsAMD64()
NewDefaultArch(runtime.GOARCH).IsARM64()
```

Reply:
Something like this?
```
func (d *DefaultArch) IsPPC64() bool {
     if d.Architecture == "ppc64le" {
           return true
     }
     return false
}

func (d *DefaultArch) GetDefaultMachinesForArch() (string, string) {
      if d.IsPPC64 {
            return DefaultPPC64LEMachineType, DefaultPPC64LEEmulatedMachines
      }
      if d.IsARM64 {
            ...
      }
      ...
}
```

Reply:
Sounds reasonable :+1: 

And now checking it again, I think your first suggestion makes a lot of sense too, since you can then probably eliminate all if/else from the general function. I did not get that. So whatever works best for you :)